NExtends/Primitives/Types/Type.extensions.cs

```diff
@@ -129,26 +129,15 @@ public static bool IsSubclassOfInterface(this Type toCheck, Type interfaceType)
 			return false;
 		}
 
-		/// <summary>
-		/// Si c'est une entité de l'EDMX, le type va être Utilisateur_DFDKKDFHVBJTHERU : Utilisateur => Utilisateur
-		/// Et parfois sera tronqué !! PublicHolidaysCalenda_SDOIDFOIDF : PublicHolidaysCalendar => PublicHolidaysCalendar !!
-		/// Ou alors des types hérité, comme FiggoApplication : Application => Application
-		/// Si c'est une APICollection, le type va être APICollection`1 => APICollection
-		/// </summary>
-		/// <param name="type"></param>
-		/// <returns></returns>
-		public static string GetRealTypeName(this Type type)
-		{
-			var info = type.GetTypeInfo();
-			if (info.IsDefined(typeof(JsonObjectAttribute), false))
-			{
-				return (info.BaseType ?? type).Name; //Au cas où BaseType est null, on prend le type lui même (cas du type Object par ex)
-			}
-			else
-			{
-				return type.Name.Split('`')[0];
-			}
-		}
+        /// <summary>
+        /// Returns IEnumerable when Type is IEnumerable of User
+        /// </summary>
+        /// <param name="type"></param>
+        /// <returns></returns>
+        public static string GetRealTypeName(this Type type)
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Consider using more general type 'System.Reflection.MemberInfo' instead of 'System.Type'. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=csharpsquid%3AS3242)

Reply:
@rducom là c'est intéressant, il me suggère d'utiliser MemberInfo plutôt que Type, mais du coup ça propage MemberInfo à GetRealTypeName( ), qui lui est très fortement lié aux Types, càd que les autre classes dérivées de MemberInfo n'ont pas "besoin" de GetRealTypeName( ).

Du coup je fais quoi ?

Reply:
Tu laisse comme ça :) Là il propose la méthode qui est sur la base class par default. Mais on veut vraiment que celà concerne un Type, donc c'est bon